### DISCUSSION TITLE
Bun v0.5.2

### DISCUSSION THREAD

**Discussion by Jarred-Sumner** [oven-sh/bun#1946] [Announcements]
**[Read Bun v0.5.2's release notes on Bun's blog](https://bun.sh/blog/bun-v0.5.2)**

This release is a follow up to [Bun v0.5](https://github.com/oven-sh/bun/releases/tag/bun-v0.5.0) and [Bun v0.5.1](https://github.com/oven-sh/bun/releases/tag/bun-v0.5.1)

To install:
```bash
curl https://bun.sh/install | bash
# or you can use npm
# npm install -g bun
```

To upgrade:
```bash
bun upgrade
```

<a href="https://bun.sh/blog/bun-v0.5.2">

<img src="https://user-images.githubusercontent.com/709451/215659858-ddb2c7ce-086a-46cc-ab37-c0dcf2abe17e.png" />

</a>

## What's Changed
* constructor parameter properties are lowered in class expressions by @dylan-conway in https://github.com/oven-sh/bun/pull/1867
* [semver] parse `^` & `~` expressions correctly by @alexlamsl in https://github.com/oven-sh/bun/pull/1854
* minor clean-ups by @alexlamsl in https://github.com/oven-sh/bun/pull/1869
* feat(dns) resolveSrv by @cirospaciari in https://github.com/oven-sh/bun/pull/1870
* enhancement(fetch): Merge parameters from request parameter with the second parameter for fetch, move verbose and proxy options to second parameter, add non-TLS tests for fetch by @cirospaciari in https://github.com/oven-sh/bun/pull/1862
* fix constructor statement order by @dylan-conway in https://github.com/oven-sh/bun/pull/1883
* fix child process node test hang by @dylan-conway in https://github.com/oven-sh/bun/pull/1884
* fix(Buffer) Fix arguments in buffer.write, fix size returned from buffer.write for utf16, fix size calc for base64, fix calc for hex byte size by @cirospaciari in https://github.com/oven-sh/bun/pull/1881
* `npm install bun` by @Electroid in https://github.com/oven-sh/bun/pull/1874
* support `bun link` of scoped packages by @alexlamsl in https://github.com/oven-sh/bun/pull/1892
* support GitHub URLs as dependencies by @alexlamsl in https://github.com/oven-sh/bun/pull/1875
* Add FileSystemRouter + React example by @scally in https://github.com/oven-sh/bun/pull/1894
* parse package-spec from CLI correctly by @alexlamsl in https://github.com/oven-sh/bun/pull/1895
* fix(WebSocket) Fix large packages receive for WS client by @cirospaciari in https://github.com/oven-sh/bun/pull/1909
* fix websocket hang by @dylan-conway in https://github.com/oven-sh/bun/pull/1910
* Implement all pending resolve methods in DNS by @cirospaciari in https://github.com/oven-sh/bun/pull/1903
* fix(Express.js) Express.js try to use function as hostname by @cirospaciari in https://github.com/oven-sh/bun/pull/1914
* ensure name is allocated with `toSliceClone` by @dylan-conway in https://github.com/oven-sh/bun/pull/1917
* [WIP] append GitHub package after fully parsed by @alexlamsl in https://github.com/oven-sh/bun/pull/1911
* fix(bench): Fix if condition always being true by @u9g in https://github.com/oven-sh/bun/pull/1923
* fix corner cases with aliased dependencies by @alexlamsl in https://github.com/oven-sh/bun/pull/1927
* normalise `bun add` package specifiers by @alexlamsl in https://github.com/oven-sh/bun/pull/1924
* parse `git+https://github.com/` as GitHub URLs by @alexlamsl in https://github.com/oven-sh/bun/pull/1926
* Support running WASI (WebAssembly) files using `bun run` by @Jarred-Sumner in https://github.com/oven-sh/bun/pull/1929
* [bun add] fix more corner cases by @alexlamsl in https://github.com/oven-sh/bun/pull/1930
* Update README.md by @Ygnys in https://github.com/oven-sh/bun/pull/1934
* [bunx] fix version parsing by @alexlamsl in https://github.com/oven-sh/bun/pull/1937
* report invalid input file as test failure by @alexlamsl in https://github.com/oven-sh/bun/pull/1938
* fix `assert()` crash by @alexlamsl in https://github.com/oven-sh/bun/pull/1941
* fix utf16le fill and utf8 partial write of utf16 by @cirospaciari in https://github.com/oven-sh/bun/pull/1943

## New Contributors
* @scally made their first contribution in https://github.com/oven-sh/bun/pull/1894
* @Ygnys made their first contribution in https://github.com/oven-sh/bun/pull/1934

**Full Changelog**: https://github.com/oven-sh/bun/compare/bun-v0.5.1...bun-v0.5.2

<hr /><em>This discussion was created from the release <a href='https://github.com/oven-sh/bun/releases/tag/bun-v0.5.2'>Bun v0.5.2</a>.</em>

**Comment by ClementNerma** [comment]
Thanks for the release!

I'm sure there's something I misunderstand here, but when I look at the reported performance improvements for `Buffer`, something seems weird. Going from ~34 ns/iter to ~77 ns/iter in the new version isn't a 2x improvement, it's more of a 2x slowdown right?

I suppose there's something I misunderstand but I can't wrap my head around this, so if you could please enlighten me ^^"

![image](https://user-images.githubusercontent.com/9782521/215715412-7de53b66-2612-405d-8a93-8fa0447e2653.png)

**Reply by Jarred-Sumner** [reply]
I got the results backwards when I copy pasted it from the terminal. The snippet has a bunch of other benchmarks in it that were irrelevant to Buffer.from   

You can run the file "buffer-create.mjs" in bench/snippets to test yourself

**Reply by ClementNerma** [reply]
I see, thanks for the explanation :)
